### PR TITLE
feat: job menu and job list structure

### PR DESCRIPTION
**Pull request by makersmelx**

### CODE REVIEW

**Comment by makersmelx** [comment]
@kendebacker @ericfu666  https://refugee-one-368a9--pr16-job-menu-and-view-fr-k7292g4v.web.app/
when you log in, take a look at `View all jobs`

File: src/components/job/JobList.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { useRequest } from "ahooks";
+import { useEffect, useMemo } from "react";
+import { Link, useParams } from "react-router-dom";
+
+import { useAuth, useJob } from "@/models";
+
+export default function JobList() {
+  const { listJobs } = useJob();
+  const { jobId } = useParams();
+  const { run, data } = useRequest(async () => listJobs(), {
```
Comment (kendebacker) [inline src/components/job/JobList.jsx:10]: Eventually the ability to fetch QueryConstraint be added to useAuth to control what output list each other gets?

**Review by kendebacker — APPROVED** [review]
Wow, nice!

File: src/components/job/JobList.jsx (hunk 1 above)
Comment (makersmelx) [reply src/components/job/JobList.jsx:10]: the user account will limit what he can see. like a client cannot see pending jobs

Of what he can see, we may still still need a filter. like he want to see jobs with keyword `counter`, that is what `queryConstraints` does.

Comment (kendebacker) [reply src/components/job/JobList.jsx:10]: So the user account will contribute parameters to queryConstraints? Or will there be separate jobList methods for each type of user?

Comment (makersmelx) [reply src/components/job/JobList.jsx:10]: no.
the account decides the whole collection of jobs he can see.
then use query to filter on the limited collection

Comment (kendebacker) [reply src/components/job/JobList.jsx:10]: Oh I see, thanks!

**Pull request merged**